source/common/stats/stats_impl.cc

```diff
@@ -136,9 +136,12 @@ bool TagExtractorImpl::extractTag(const std::string& stat_name, std::vector<Tag>
 }
 
 RawStatData* HeapRawStatDataAllocator::alloc(const std::string& name) {
-  uint64_t num_bytes_to_allocate = RawStatData::sizeGivenName(name);
+  uint64_t num_bytes_to_allocate = RawStatData::structSize(name.size());
   RawStatData* data = static_cast<RawStatData*>(::calloc(num_bytes_to_allocate, 1));
-  data->initialize(name, num_bytes_to_allocate);
+  if (data == nullptr) {
+    throw EnvoyException("HeapRawStatDataAllocator: unable to allocate a new stat");
```

Comment:
This might be caught. Can you throw some other type of exception? (Like whatever type new() throws).

Reply:
I changed it to `std::bad_alloc()` in 4cad816.